### PR TITLE
Refactor AntiJoin

### PR DESCRIPTION
**Pull request by zengyan1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/sql/colexec/anti/join.go [hunk 1]
```diff
@@ -135,30 +133,32 @@ func (antiJoin *AntiJoin) Call(proc *process.Process) (vm.CallResult, error) {
...
-	if ctr.rbat != nil {
-		proc.PutBatch(ctr.rbat)
-		ctr.rbat = nil
-	}
-	ctr.rbat = batch.NewWithSize(len(ap.Result))
-	for i, pos := range ap.Result {
-		ctr.rbat.Vecs[i] = proc.GetVector(*ap.ctr.bat.Vecs[pos].GetType())
-		// for anti join, if left batch is sorted , then output batch is sorted
-		ctr.rbat.Vecs[i].SetSorted(ap.ctr.bat.Vecs[pos].GetSorted())
+	anal.Input(ap.ctr.inbat, isFirst)
```
Comment (ouyuanning) [inline pkg/sql/colexec/anti/join.go:147]: 这一段对于emptyProbe或者Probe代码应该是一样的，
感觉可以提取到 Call函数中 case Probe的部分

File: pkg/sql/colexec/anti/join.go [hunk 2]
```diff
@@ -95,34 +97,30 @@ func (antiJoin *AntiJoin) Call(proc *process.Process) (vm.CallResult, error) {
...
 				}
-
-				ap.ctr.bat = bat
+				ap.ctr.inbat = bat
 				ap.ctr.lastrow = 0
 			}
 
 			if ctr.mp == nil {
-				err := ctr.emptyProbe(ap, proc, anal, antiJoin.GetIsFirst(), &result)
+				err = ctr.emptyProbe(ap, proc, anal, antiJoin.GetIsFirst(), &probeResult)
```
Comment (fengttt) [inline pkg/sql/colexec/anti/join.go:105]: This API is simply wrong.    in the emptyProbe, there are still PutBatch

File: pkg/sql/colexec/anti/join.go (hunk 1 above)
Comment (zengyan1) [reply pkg/sql/colexec/anti/join.go]: 已修改

File: pkg/sql/colexec/anti/join.go [hunk 3]
```diff
@@ -95,34 +97,41 @@ func (antiJoin *AntiJoin) Call(proc *process.Process) (vm.CallResult, error) {
 				if bat.IsEmpty() {
 					continue
 				}
-
-				ap.ctr.bat = bat
+				ap.ctr.inbat = bat
```
Comment (fengttt) [inline pkg/sql/colexec/anti/join.go:100]: Do you really need the inbat?   Seems useless

File: pkg/sql/colexec/anti/join.go [hunk 4]
```diff
@@ -95,34 +97,41 @@ func (antiJoin *AntiJoin) Call(proc *process.Process) (vm.CallResult, error) {
...
 			}
 
-			if ctr.mp == nil {
-				err := ctr.emptyProbe(ap, proc, anal, antiJoin.GetIsFirst(), &result)
-				if err != nil {
-					return result, err
+			if ap.ctr.lastrow == 0 {
+				anal.Input(ap.ctr.inbat, antiJoin.GetIsFirst())
+			}
+			if ctr.rbat == nil {
```
Comment (fengttt) [inline pkg/sql/colexec/anti/join.go:107]: You can put ctr.rbat allocation in Prepare and never worry about this in Call.   Simple CleanOnlyData in Call.

**Review by fengttt — COMMENTED** [review]
My comment may not be relevant.    I am not that sure about join code.    Up to your own judgement.

File: pkg/sql/colexec/anti/join.go (hunk 3 above)
Comment (zengyan1) [reply pkg/sql/colexec/anti/join.go]: 是的，antijoin的多次probe不会使用同一个inbat，已删除

File: pkg/sql/colexec/anti/join.go (hunk 4 above)
Comment (zengyan1) [reply pkg/sql/colexec/anti/join.go:101]: 已修改，innerjoin遗漏此处也已修改

**Pull request merged**